packages/story-editor/src/components/form/mediaUploadButton.js

```diff
@@ -127,7 +121,10 @@ function MediaUploadButton({ onInsert, renderButton, buttonInsertText }) {
     (resource) => {
       try {
         if (isTranscodingEnabled && canTranscodeResource(resource)) {
-          if (transcodableMimeTypes.includes(resource.mimeType)) {
+          if (
+            !resource.isOptimized &&
+            TRANSCODABLE_MIME_TYPES.includes(resource.mimeType)
+          ) {
```

Comment:
Explanation for this change:

`transcodableMimeTypes` did not include mime types that were allowed for upload, i.e. `.mp4` or `.webm`. So such files were never optimized.

With this change, `.mp4` or `.webm` files now also get passed to `optimizeVideo`, which then checks whether the file actually needs to be optimized/transcoded.

Reply:
Could we do this ? https://github.com/GoogleForCreators/web-stories-wp/pull/11454/files#r909766216

Reply:
I don't think so. `allowedVideoMimeTypes` could theoretically contain mime types that FFmpeg does not support (i.e. are not in `TRANSCODABLE_MIME_TYPES `). However, we only want to allow videos that can be transcoded if needed. Otherwise `optimizeVideo()` could error.